### DISCUSSION TITLE
Do you plan to support MySQL/MariaDB?

### DISCUSSION THREAD

**Discussion by maliayas** [mathesar-foundation/mathesar#3447] [Feature Ideas]
I've searched everywhere and found no reference to this. So asking it here.

**Comment by kgodey** [comment]
@maliayas Thanks for your question! We don't have any current plans to support MySQL/MariaDB or any database other than PostgreSQL at this time. Having a backend that's generic / abstract enough to support multiple databases will make it hard for us to have the deep integration with the DB layer that we're aiming for.

We do plan to support [foreign data wrappers](https://wiki.postgresql.org/wiki/Foreign_data_wrappers), though, and you can use MySQL/MariaDB that way once we do. See [this discussion](https://github.com/mathesar-foundation/mathesar/discussions/3414) for more context.

Once we're satisfied with how Mathesar works with Postgres, we will consider other DB backends, but I don't expect that project to start for a couple of years.

**Reply by maliayas** [reply]
Ah I see. Thanks for the reply. Just subscribed to that discussion.

**Reply by kgodey** [reply]
You're welcome!

**Reply by shantanuo** [reply]
mysql support will be very useful. I hope you will consider this request.
